internal/app/machined/pkg/runtime/v1alpha1/platform/aws/metadata.go

```diff
@@ -7,9 +7,10 @@ package aws
 import (
 	"context"
 	"fmt"
-	"net/http"
+	"io"
 
-	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+	"github.com/aws/smithy-go"
```

Comment:
at first thought this was something by Spencer :joy:

Reply:
lol, that was my initial thought as well... why smithy? and it turns out to be an HTTP client on steroids